pkg/operator/deploy/deploy.go

```diff
@@ -196,6 +196,33 @@ func (o *operator) resources(ctx context.Context) ([]kruntime.Object, error) {
 	}
 
 	// then dynamic resources
+	if o.oc.UsesWorkloadIdentity() {
+		var operatorIdentity *api.PlatformWorkloadIdentity // use a pointer to make it easy to check if we found an identity below
+		for _, i := range o.oc.Properties.PlatformWorkloadIdentityProfile.PlatformWorkloadIdentities {
+			if i.OperatorName == pkgoperator.OperatorIdentityName {
+				operatorIdentity = &i
```

Comment:
Should we break here?

```suggestion
				operatorIdentity = &i
				break
```